run-tests.sh

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/env bash
+
+set -euo pipefail
+
+cd "$(dirname "$0")"
+
+port=9000
+containers=()
+network=$(LC_ALL=C tr -dc 'a-z' </dev/urandom | head -c 13 ; echo)
+
+print_usage() {
+  cat <<EOF
+usage: $0 [IMAGE...]
+
+examples:
+       $0 "sonar-scanner-cli:4.1"
+EOF
+}
+
+info() {
+  echo "[info] $@"
+}
+
+warn() {
+  echo "[warn] $@" >&2
+}
+
+fatal() {
+  echo "[error] $@" >&2
+  exit 1
+}
+
+require() {
+  local prog missing=()
+  for prog; do
+    if ! type "$prog" &>/dev/null; then
+      missing+=("$prog")
+    fi
+  done
+
+  [[ ${#missing[@]} == 0 ]] || fatal "could not find required programs on the path: ${missing[@]}"
+}
+
+create_network() {
+  docker network create "$network"
+}
+
+destroy_network() {
+  docker network rm "$network"
+}
+
+wait_for_sonarqube() {
+  local i web_up=no sonarqube_up=no
+
+  for ((i = 0; i < 10; i++)); do
+    info "waiting for web server to start ..."
+    if curl -sI localhost:$port | grep '^HTTP/.* 200'; then
+      web_up=yes
+      break
+    fi
+    sleep 5
+  done
+
+  [[ $web_up == yes ]] || return 1
+
+  for ((i = 0; i < 20; i++)); do
+    info "waiting for sonarqube to be ready ..."
+    if curl -s localhost:$port/api/system/status | grep '"status":"UP"'; then
+      sonarqube_up=yes
+      break
+    fi
+    sleep 10
+  done
+
+  [[ "$sonarqube_up" == yes ]]
+}
+
+test_scanner() {
+  local scanner_finished_successfuly=no container_name
+  container_name=$(LC_ALL=C tr -dc 'a-z' </dev/urandom | head -c 13 ; echo)
+  echo "testing image $1 in container $container_name"
+
+  git clone https://github.com/SonarSource/sonar-scanning-examples.git "$container_name"
+  echo "sonar.projectKey=$container_name-test" >>"$(pwd)/$container_name/sonarqube-scanner/sonar-project.properties"
+  echo "sonar.host.url=http://${containers[0]}:9000" >>"$(pwd)/$container_name/sonarqube-scanner/sonar-project.properties"
+
+  docker run --network="$network" --name="$container_name" -it -v "$(pwd)/$container_name/sonarqube-scanner:/usr/src" "$1"
+  containers+=("$container_name")
```

Comment:
Cosmetic

Isn't a practice to use upper case variable names for global one and lower case variable names for local ones?

In this case, it would help readability

Reply:
I'm not aware of any such convention but I'm not a bash guru ;) [This](https://stackoverflow.com/questions/32144204/what-is-a-convention-for-naming-a-constant-in-bash) says the capital letters in vars should be used for env vars.

Reply:
I'm no guru either.

This or anything similar would be convenient to distinguish between the local and global variable. I don't know whether there is any such accepted convention (invoking @janos-ss )

nevertheless, this is minor and local variables are declared only a couple of lines above, this is readable enough (I managed to figure the scope of variables, who need a better proof that it is enough? ;p)

Reply:
Thanks @sns-seb for invoking me.

I share the opinion to not create variables with all-caps naming, to reserve that to environment variables, exported by system tools. This is the style I try to follow when creating new scripts at SonarSource.

This is not a standard though, for example [Google's style guide](https://google.github.io/styleguide/shell.xml#Naming_Conventions) permits all-caps for "constants".

Reply:
thanks for sharing knowledge, @janos-ss